Etabs_Adapter/CRUD/Read/Bar.cs

```diff
@@ -126,6 +126,15 @@ private List<Bar> ReadBar(List<string> ids = null)
                         etabsIdFragment.Story = story;
                     }
 
+                    // Get the groups the bar is assigned to
+                    int numGroups = 0;
+                    string[] groupNames = new string[0];
+                    if (m_model.FrameObj.GetGroupAssign(id, ref numGroups, ref groupNames) == 0)
+                    {
+                        foreach (string grpName in groupNames)
+                            bhBar.Tags.Add(grpName);
+                    }
+
```

Comment:
Works well with Etabs 17 and onwards, but the functionality and API call does not exist for Etabs2016. We need to add the following pre-compiler directives to avoid this being part of the 2016 build:

Please feel free to just commit the suggestion in this comment, or handle it elsewhere if you like.

Please be advised that I had to rebase and force push the previous commit on the branch due to conflict :)

```suggestion
#if !Debug16 && !Release16  //Not supported in ETABS 2016
                    // Get the groups the bar is assigned to
                    int numGroups = 0;
                    string[] groupNames = new string[0];
                    if (m_model.FrameObj.GetGroupAssign(id, ref numGroups, ref groupNames) == 0)
                    {
                        foreach (string grpName in groupNames)
                            bhBar.Tags.Add(grpName);
                    }
#endif
```